apps/server/src/modules/deletion/repo/scope/deletion-request-scope.spec.ts

```diff
@@ -34,30 +34,63 @@ describe(DeletionRequestScope.name, () => {
 
 	describe('byStatusAndDate', () => {
 		const setup = () => {
-			const newerThan = new Date(Date.now() - 15 * 60 * 1000); // 15 minutes ago
-			const olderThan = new Date(Date.now() - 5 * 60 * 1000); // 5 minutes ago
+			const fifteenMinutesAgo = new Date(Date.now() - 15 * 60 * 1000);
+			const fiveMinutesAgo = new Date(Date.now() - 5 * 60 * 1000);
 
 			const expectedQueryOlder = {
 				status: {
 					$in: [StatusModel.FAILED, StatusModel.PENDING],
 				},
-				updatedAt: {
-					$lt: olderThan,
-				},
+				$and: [
+					{
+						updatedAt: {
+							$lt: fiveMinutesAgo,
+						},
+					},
+				],
 			};
 			const expectedQueryNewer = {
 				status: {
 					$in: [StatusModel.FAILED, StatusModel.PENDING],
 				},
-				updatedAt: {
-					$gte: newerThan,
+				$and: [
+					{
+						updatedAt: {
+							$gte: fifteenMinutesAgo,
+						},
+					},
+				],
+			};
+			const expectedQueryOlderAndNewer = {
+				status: {
+					$in: [StatusModel.FAILED, StatusModel.PENDING],
 				},
+				$and: [
+					{
+						updatedAt: {
+							$lt: fiveMinutesAgo,
+						},
+					},
+					{
+						updatedAt: {
+							$gte: fifteenMinutesAgo,
+						},
+					},
+				],
 			};
+			const expectedQueryNoDates = {
+				status: {
+					$in: [StatusModel.FAILED, StatusModel.PENDING],
```

Comment:
what about filter for `updatedAt: null` ?

Reply:
we have the case tested at line 126